RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc

```diff
@@ -557,23 +564,15 @@ void PulseShapeFitOOTPileupCorrection::phase1Apply(const HBHEChannelInfo& channe
 
     // Photo statistics uncertainties
     //      sigmaFC/FC = 1/sqrt(Ne);
+    // Note2. (from kPedro): the output number of photoelectrons after smearing is treated very differently for SiPMs: *each* pe is assigned a different time based on a random generation from the Y11 pulse plus the SimHit time. In HPDs, the overall pulse is shaped all at once using just the SimHit time.
+
     noisePHArr[ip] = 0;
     if(channelData.hasTimeInfo() && (charge-ped)>channelData.tsPedestalWidth(ip)) {
       noisePHArr[ip] = sqrt((charge-ped)*channelData.fcByPE());
     }
-
-    /*
-    // FIXME: in the future switch the photostatistics on also for the HPD
-    if(!channelData.hasTimeInfo() && (charge-ped)>channelData.tsPedestalWidth(ip)) {
-
-      // Note1. the fcByPE is not in the DB, but is hard coded in python
-      // https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_0/SimCalorimetry/HcalSimProducers/python/hcalSimParameters_cfi.py#L62
-
-      // Note2. (from kPedro): the output number of photoelectrons after smearing is treated very differently for SiPMs: *each* pe is assigned a different time based on a random generation from the Y11 pulse plus the SimHit time. In HPDs, the overall pulse is shaped all at once using just the SimHit time.
-
-      noisePHArr[ip] = sqrt((charge-ped)*0.3305);
+    if(!channelData.hasTimeInfo() && (charge-ped)>noise_) {
+      noisePHArr[ip] = sqrt((charge-ped)*channelData.fcByPE());
```

Comment:
the amazing class HBHEChannelInfo channelData.fcByPE() is only available for the Phase1Code.

For the run1/run2 code I will likely have to do a lot of gymnastic.
I tried to do this in the past unsuccessfully for the other parameters i.e. the noise https://github.com/mariadalfonso/cmssw/commit/c923f24e48c922581aeee84422af9e8db29cb466
The old M2 code is anyway full of all hard coded numbers so one more or less should not make a big difference. I can see if I can find an easy way, but no promises (maybe we can put in the post-pre5 SW rework)?